interop/tapir/src/main/scala/caliban/interop/tapir/RequestInterceptor.scala

```diff
@@ -8,17 +8,20 @@ import zio.ZIO
  * RequestInterceptor provides a way to extract context from the http request, potentially failing before
  * query execution or injecting context into ZIO environment.
  */
-trait RequestInterceptor[-R] { self =>
-  def apply[R1 <: R, A](request: ServerRequest)(e: ZIO[R1, TapirResponse, A]): ZIO[R1, TapirResponse, A]
+trait RequestInterceptor[-R, +DynR] { self =>
+  def apply[R1 <: R, A](request: ServerRequest)(e: ZIO[R1 with DynR, TapirResponse, A]): ZIO[R1, TapirResponse, A]
 
-  def |+|[R1 <: R](that: RequestInterceptor[R1]): RequestInterceptor[R1] = new RequestInterceptor[R1] {
-    override def apply[R2 <: R1, A](request: ServerRequest)(e: ZIO[R2, TapirResponse, A]): ZIO[R2, TapirResponse, A] =
-      that.apply[R2, A](request)(self.apply[R2, A](request)(e))
-  }
+  def |+|[R1 <: R, DynR2](that: RequestInterceptor[R1, DynR2]): RequestInterceptor[R1, DynR with DynR2] =
+    new RequestInterceptor[R1, DynR with DynR2] {
+      override def apply[R2 <: R1, A](request: ServerRequest)(
+        e: ZIO[R2 with DynR with DynR2, TapirResponse, A]
+      ): ZIO[R2, TapirResponse, A] =
+        that.apply[R2, A](request)(self.apply[R2 with DynR2, A](request)(e))
+    }
 }
 
 object RequestInterceptor {
-  val empty: RequestInterceptor[Any] = new RequestInterceptor[Any] {
-    override def apply[R, A](request: ServerRequest)(e: ZIO[R, TapirResponse, A]): ZIO[R, TapirResponse, A] = e
+  val empty: RequestInterceptor[Any, Any] = new RequestInterceptor[Any, Any] {
+    override def apply[R, A](request: ServerRequest)(e: ZIO[R with Any, TapirResponse, A]): ZIO[R, TapirResponse, A] = e
```

Comment:
Can remove `with Any`

Reply:
2.12 as well as 2.13, and changing the context bound like you suggest sadly doesn't solve it.

Reply:
Might be a bug in `scalac` (I'm guessing Scala 3 allows this right?) or that some magical import / implicit / compiler flag is required 😓. I haven't used Scala 2 in a while so I'm not sure what the solution is here; although if using `R with Any` gets it to work then I'm guessing that's fine?

Reply:
I'm actually unsure how I get to compile the interop module with Scala 3. Setting the whole build to Scala 3 fails because `sbt-scripted` isn't available.

Reply:
I usually just run this https://github.com/ghostdogpr/caliban/blob/series/2.x/.circleci/config.yml#L97 in the terminal 😅

Reply:
Hmm that's weird, I never had this issue for example with ZIO environment. It could be quite confusing for users if they have to do this to make it work 🤔